protocol/client.go

```diff
@@ -1,46 +1,51 @@
 package protocol
 
 import (
+	"context"
+	"io"
+
 	"github.com/domainr/epp2/protocol/dataunit"
 	"github.com/domainr/epp2/schema"
 	"github.com/domainr/epp2/schema/epp"
 )
 
-// Client is a low-level client for the Extensible Provisioning Protocol (EPP)
-// as defined in [RFC 5730]. A Client is safe to use from multiple goroutines.
+// Client represents a low-level EPP client as defined in [RFC 5730].
+// A Client is safe to use from multiple goroutines.
 //
 // [RFC 5730]: https://datatracker.ietf.org/doc/rfc5730/
 type Client interface {
 	// ExchangeEPP sends an EPP message and returns an EPP response.
-	// It blocks until a response is received, ctx is canceled, or
+	// It blocks until a response is received, the Context is canceled, or
 	// the underlying connection is closed.
-	ExchangeEPP(epp.Body) (epp.Body, error)
-
-	// Close closes the connection.
-	Close() error
+	ExchangeEPP(context.Context, epp.Body) (epp.Body, error)
 }
 
 type client struct {
 	client dataunit.Client
 	coder  coder
 }
 
-// Connect connects to an EPP server over conn. It waits for the initial
-// <greeting> message from the server before returning.
-// Responses from the server will be decoded using [schemas.Schema] schemas.
+// Connect connects to an EPP server over conn. It blocks until the initial
+// <greeting> message is received, ctx is canceled, or the underlying connection is closed.
+//
+// The supplied Context must be non-nil, and only affects reading the initial <greeting> from the client.
+// Canceling ctx expires after Connect returns does not affect the resulting Client.
+//
+// Responses from the server will be decoded using [schema.Schema] schemas.
 // If no schemas are provided, a set of reasonable defaults will be used.
-func Connect(conn dataunit.Conn, schemas ...schema.Schema) (Client, epp.Body, error) {
+func Connect(ctx context.Context, conn io.ReadWriter, schemas ...schema.Schema) (Client, epp.Body, error) {
```

Comment:
```suggestion
func Connect(greetingCtx context.Context, conn io.ReadWriter, schemas ...schema.Schema) (Client, epp.Body, error) {
```

I wonder if there's a better name than `Connect` for this function since it is really only about receiving the greeting, which initializes the EPP protocol over whatever connection is already established in `conn`.

Reply:
It initializes the `Client`, which maintains the schemas, but to your point…does it need to carry that state?